opm/autodiff/BlackoilPropsAdFromDeck.cpp

```diff
@@ -170,23 +170,24 @@ BlackoilPropsAdFromDeck::BlackoilPropsAdFromDeck(const BlackoilPropsAdFromDeck&
         waterPvt_->initFromDeck(deck, eclState);
 
         // Surface densities. Accounting for different orders in eclipse and our code.
-        const auto& densityKeyword = deck.getKeyword("DENSITY");
-        int numRegions = densityKeyword.size();
-        auto tables = eclState.getTableManager();
+        const auto& densities = eclState.getTableManager().getDensityTable();
+        const auto numRegions = densities.size();
+        const auto& tables = eclState.getTableManager();
```

Comment:
Looks fine, just a small detail: now that we get the densities from the table manager, what is now in line 173 should be below this line, and use `tables`.